src/convertToObject.js

```diff
@@ -1,18 +1,25 @@
 'use strict';
 
 /**
- * Implement convertToObject function:
- *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
- * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
- *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const cssStyles = {};
+
+  const redactedString = sourceString
+    .split(';')
+    .filter(s => s.trim().length > 0)
+    .join(':')
+    .split(':')
+    .map(s => s.trim());
+
+  for (let i = 0; i < redactedString.length; i += 2) {
+    cssStyles[redactedString[i]] = redactedString[i + 1];
+  }
```

Comment:
```suggestion
    .forEach(prop => {
      const [ propName, propValue ] = prop.split(':');
      cssStyles[propName.trim()] = propValue.trim();
    });
```

Array has nice iteration method `forEach`. Use it here, cause you need to iterate through filtered lines.
It can optimize your code.